client.go

```diff
@@ -2907,3 +2902,75 @@ func releasePipelineWork(pool *sync.Pool, w *pipelineWork) {
 	w.err = nil
 	pool.Put(w)
 }
+
+// Resource Clean interface.
+// resourceClean can register into clientCleaner for cleaning resources.
+type resourceClean interface {
+	cleanResource()
+	hasResource() bool
+}
+
+// Cleaner to clean resources.
+// Client, HostClient register into cleaner. The Cleaner will check and delete client
+// where it has no resources.
+type clientCleaner struct {
+	initOnce       sync.Once
+	// clients to clean
+	clients        sync.Map
+}
+
+// Global cleaner.
+var cCleaner = &clientCleaner{}
+
+// Initialize cleaner.
+func (c *clientCleaner) init() {
+	c.initOnce.Do(func() {
+		go c.cleaner()
+	})
+}
+
+// Clean clients
+func (c *clientCleaner) cleaner() {
+	for {
+		c.cleanClient()
+		time.Sleep(10 * time.Second)
+	}
+}
+
+// Register Client to cleaner.
+func (c *clientCleaner) register(client resourceClean) {
+	if client == nil {
+		return
+	}
+	// Check if init
+	c.init()
+
+	// Store client
```

Comment:
```suggestion
```
The statement below is already super clear, it doesn't need a comment like this.